### DISCUSSION TITLE
Inkstich scripting

### DISCUSSION THREAD

**Discussion by ricardoquesada** [inkstitch/inkstitch#3219] [Q&A]
Hi,
I want to automate part of the digitizing process.
For example, I want squares to use a certain fill-stitch, and circles a different one, and so on.

Is there an "scripting API" to do that ?

Or what could be a possible solution ? Thanks!

**Comment by tyrosinase** [comment]
I do something like that, but generally with running stitches and stroke style (one dash pattern for bean stitch, a sparser one for basting stitch, etc.) What I do is just edit the .svg file externally. It's just XML, so almost every language has a library that will handle it, though I admit that since I'm only processing my own files, I just hardcode what it's looking for. (Laziness is a virtue to a Perl coder!)

**Comment by ricardoquesada** [comment]
good to know.
yes,  autogenerating the .svg file should work. I thought of doing that, but was hoping for something with fewer steps.
But definitely something that I can do.

**Comment by tyrosinase** [comment]
It *should* be possible to do some of that on the command line but the command-line stuff is poorly-documented at best. I do some resizing and rotating of entire designs that way, but haven't messed around too much with manipulating individual objects. You can select all the objects of a type, but I'm not sure if you can apply non-standard attributes.

That's technically all an Inkscape thing rather than Ink/Stitch, though.
